backend/tracim_backend/lib/webdav/dav_provider.py

```diff
@@ -1,33 +1,140 @@
+                    root_workspace_filemanager_filename
+                )
+            except WorkspaceNotFound:
+                root_workspace = None
+            self.workspaces.append(root_workspace)
+            # INFO - G.M - 2020-10-07 - Get all workspace tree + content_tree
+            if root_workspace:
+                search_sub_workspace = True
+                for path_part in path_parts[1:]:
+                    section = webdav_convert_file_name_to_bdd(path_part)
+                    if search_sub_workspace:
+                        try:
+                            workspace = self.workspace_api.get_one_by_filemanager_filename(
+                                section, parent=self.workspaces[-1]
+                            )
+                            self.workspaces.append(workspace)
+                            continue
+                        except WorkspaceNotFound:
+                            search_sub_workspace = False
+                            pass
+                    try:
+                        content = self.content_api.get_one_by_filename(
+                            filename=section,
+                            workspace=self.workspaces[-1],
+                            parent=self.contents[-1] if self.contents else None,
+                        )
+                    except ContentNotFound:
+                        content = None
+                    self.contents.append(content)
+
+    def _path_splitter(self, path: str) -> typing.List[str]:
+        path_parts = path.split("/")
+        # removing leading or trailing empty path part
+        if not path_parts[0]:
+            path_parts = path_parts[1:]
+        if path_parts and not path_parts[-1]:
+            path_parts = path_parts[:-1]
+        return path_parts
+
+    @property
+    def current_workspace(self) -> typing.Optional[Workspace]:
+        try:
+            return self.workspaces[-1]
+        except IndexError:
+            return None
+
+    @property
+    def current_content(self) -> typing.Optional[Content]:
+        try:
+            return self.contents[-1]
+        except IndexError:
+            return None
+
+    @property
+    def current_parent_content(self) -> typing.Optional[Content]:
+        try:
+            return self.contents[-2]
+        except IndexError:
+            return None
+
+    @property
+    def current_path_level(self) -> typing.Optional[ResourceLevelType]:
+        """
+        return the level type of the current path:
+        - return ResourceLevelType.ROOT is path is root path
+        - return ResourceLevelType.WORKSPACE if path is at workspace level (can be direct
+        workspace root or subworkspace root)
+        - return ResourceLevelType.CONTENT if path is at content level (can be direct content at workspace
+        root or content within directory hierarchy)
+        - Return None if path is not valid
+        """
+        if self.path == "/":
+            return ResourceLevelType.ROOT
```

Comment:
Maybe only use `self.workspaces`/`self.contents`:
```python
type = None
if not self.current_workspace:
   type = ResourceLevelType.ROOT
elif not self.contents:
   type = ResourceLevelType.WORKSPACE
elif self.current_content:
   type = ResourceLevelType.CONTENT
return type
```

Reply:
Doesn't work for error 404 case. self.current_workspace will be none in this case.

Reply:
another way to do this:
```python
        type = None
        if self.current_content:
            type = ResourceType.CONTENT
        elif self.current_workspace:
            type = ResourceType.WORKSPACE
        elif self.path == "/":
            type = ResourceType.ROOT
        return type
```